src/content/docs/en/guides/cms/builderio.mdx

```diff
@@ -3,19 +3,503 @@ title: Builder.io & Astro
+
+  const { results: posts } = await fetch(
+    `https://cdn.builder.io/api/v3/content/${builderModel}?${new URLSearchParams(
+      {
+        apiKey: builderAPIpublicKey,
+        fields: ["data.slug", "data.title"].join(","),
+        cachebust: "true",
+      }
+    ).toString()}`
+  )
+    .then((res) => res.json())
+    .catch
+    // ...catch some errors...);
+    ();
+
+  return [
+    ...posts.map(({ data: { slug, title } }) => [
+      {
+        params: { slug },
+        props: { title },
+      },
+    ]),
+  ];
+}
+---
+```
+
+This fetch call is the same you've used to [display a list of blog posts](#displaying-a-list-of-blog-posts). Here, the `getStaticPaths` function creates a static route for each blog post, based on the `slug` data field.
+
+Next, add the following content:
+
+```astro title="src/pages/posts/[slug].astro" ins={30-49}
+---
+export async function getStaticPaths() {
+  const builderModel = import.meta.env.BUILDER_BLOGPOST_MODEL;
+  const builderAPIpublicKey = import.meta.env.BUILDER_API_PUBLIC_KEY;
+
+  const { results: posts } = await fetch(
+    `https://cdn.builder.io/api/v3/content/${builderModel}?${new URLSearchParams(
+      {
+        apiKey: builderAPIpublicKey,
+        fields: ["data.slug", "data.title"].join(","),
+        cachebust: "true",
+      }
+    ).toString()}`
+  )
+    .then((res) => res.json())
+    .catch
+    // ...catch some errors...);
+    ();
+
+  return [
+    ...posts.map(({ data: { slug, title } }) => [
+      {
+        params: { slug },
+        props: { title },
+      },
+    ]),
+  ];
+}
+
+const { slug } = Astro.params;
+const { title } = Astro.props;
+
+const builderModel = import.meta.env.BUILDER_BLOGPOST_MODEL;
+const builderAPIpublicKey = import.meta.env.BUILDER_API_PUBLIC_KEY;
+// Builder's API requires this field but for this use case the URL doesn't seem to matter - the API returns the same HTML
+const encodedUrl = encodeURIComponent("moot");
+
+const { html: postHTML } = await fetch(
+  `https://cdn.builder.io/api/v1/qwik/${builderModel}?${new URLSearchParams({
+    apiKey: builderAPIpublicKey,
+    url: encodedUrl,
+    "query.data.slug": slug,
+    cachebust: "true",
+  }).toString()}`
+)
+  .then((res) => res.json())
+  .catch();
+---
+```
+
+The `slug` returned from `getStaticPaths` is used to fetch the corresponding post. The HTML that Builder sends back is made of the blocks you put on the page in Builder's editor.
+
+:::caution
+This code relies on the fact that each content entry in the `blogpost` model has non-empty fields called `slug` and `title`. If you didn't create a fresh model for this guide, make sure that all content entries have these fields filled, including those that you started with.
+:::
+
+:::note
+The variables `builderModel` and `builderAPIpublicKey` need to be created twice, since [`getStaticPaths` runs in its own isolated scope](/en/reference/api-reference/#getstaticpaths).
+:::
```

Comment:
```suggestion
```
I'm going to suggest we cut ALL this, and use the next code sample only and describe BOTH the new frontmatter stuff AND the html rendering.

So instead of "Finally, Render the HTML", let's beef up that content with a nod to both:
- using slug to fetch the corresponding post AND
- rendering that content

I also suggest we:
- nix the caution, as we've already established early on in prereqs AND in "create a few blog posts" what the model should look like
- combine the concepts of using slug to fetch the post again, and that you need to re-provide the builder specific variables outside of `getStaticPath`'s scope and try to make one solid explanation of what's happening with this second bit of frontmatter code.

Reply:
marking this as TODO

Reply:
Totally fine to get to this when you can! Please do ping me again to let me know that you've done this, and I should take a look again!  :raised_hands: